pkg/controller/argocd/argocd_controller.go

```diff
@@ -64,7 +66,8 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileArgoCD{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+	kc, _ := k8s.NewForConfig(mgr.GetConfig())
```

Comment:
wish to figure out a way to avoid adding a second client..

Reply:
return statement(&ReconcileArgoCD) and return type(reconcile.Reconciler) are different.